diff --git a/openid_connect.install b/openid_connect.install
index 0c1b3ce..6364375 100644
--- a/openid_connect.install
+++ b/openid_connect.install
@@ -121,8 +121,8 @@ function openid_connect_update_8105() {
   $database = \Drupal::database();
   // Get the IDs for the duplicate entries.
   $query = $database->select('openid_connect_authmap', 'a1')->fields('a1', ['aid']);
-  $query->innerJoin('openid_connect_authmap', 'a2');
-  $query->where("a1.aid > a2.aid AND a1.uid = a2.uid AND a1.client_name = a2.client_name AND a1.sub = a2.sub");
+  $query->innerJoin('openid_connect_authmap', 'a2', ('a1.uid = a2.uid AND a1.client_name = a2.client_name AND a1.sub = a2.sub'));
+  $query->where("a1.aid > a2.aid");
   $aids = $query->distinct()->execute()->fetchAllAssoc('aid');
 
   if (!empty($aids)) {
