diff --git a/modules/views/src/Plugin/views/query/Sql.php b/modules/views/src/Plugin/views/query/Sql.php
index 64b80dd..52af4ea 100644
--- a/modules/views/src/Plugin/views/query/Sql.php
+++ b/modules/views/src/Plugin/views/query/Sql.php
@@ -513,6 +513,16 @@ class Sql extends QueryPluginBase {
       $join = $this->adjustJoin($join, $relationship);
     }
 
+    // Ensure the table alias is no longer than 60 characters long.
+    $alias = (strlen($alias) > 60) ? substr($alias, -60) : $alias;
+
+    // Make sure $alias isn't already used; if it, start adding stuff.
+    $alias_base = $alias;
+    $count = 1;
+    while (!empty($this->tableQueue[$alias])) {
+      $alias = $alias_base . '_' . $count++;
+    }
+
     $this->tableQueue[$alias] = [
       'table' => $table,
       'num' => $this->tables[$relationship][$table]['count'],
