diff --git a/src/Plugin/Field/FieldWidget/Select2EntityReferenceWidget.php b/src/Plugin/Field/FieldWidget/Select2EntityReferenceWidget.php
index 0621f2e..0545530 100644
--- a/src/Plugin/Field/FieldWidget/Select2EntityReferenceWidget.php
+++ b/src/Plugin/Field/FieldWidget/Select2EntityReferenceWidget.php
@@ -205,7 +205,13 @@ class Select2EntityReferenceWidget extends Select2Widget {
     /** @var \Drupal\Core\Entity\EntityReferenceSelection\SelectionInterface $handler */
     $handler = \Drupal::service('plugin.manager.entity_reference_selection')->getInstance($handler_settings);
 
-    $options = empty($element['#options']) ? [] : static::getValidReferenceableEntities(array_keys(OptGroup::flattenOptions($element['#options'])), $handler_settings);
+    $options = [];
+    if (!empty($element['#options'])) {
+      $options = array_keys(OptGroup::flattenOptions($element['#options']));
+      $options = array_filter($options, 'is_numeric');
+    }
+    $options = empty($options) ? [] : static::getValidReferenceableEntities($options, $handler_settings);
+
     $items = [];
     foreach ($values as $value) {
       if (isset($options[$value])) {
